src/FrequencyMapAgent.cpp

```diff
@@ -169,36 +167,52 @@ namespace geopm
         }
     }
 
+    bool FrequencyMapAgent::is_all_nan(const std::vector<double> &vec)
+    {
+        return std::all_of(vec.begin(), vec.end(),
+                           [](double x) -> bool { return std::isnan(x); });
+    }
+
     void FrequencyMapAgent::update_policy(const std::vector<double> &policy)
     {
-        m_is_policy_updated = m_freq_governor->set_frequency_bounds(
-            policy[M_POLICY_FREQ_MIN], policy[M_POLICY_FREQ_MAX]);
+        if (is_all_nan(policy)) {
+            // All-NAN policy is ignored
+            m_is_policy_updated = false;
+            return;
+        }
 
+        std::map<uint64_t, double> old_freq_map = m_hash_freq_map;
+        m_hash_freq_map.clear();
         for (auto it = policy.begin() + M_POLICY_FIRST_HASH;
-             it != policy.end() && std::next(it) != policy.end(); std::advance(it, 2)) {
+             it != policy.end() && std::next(it) != policy.end();
+             std::advance(it, 2)) {
```

Comment:
Why?

Also, it looks like there's a dangling curly brace at EOL.

Reply:
Why advance two elements?  One is the hash one is the frequency.  Not sure about the dangling curly, it was there before the change.